# 4주차 회귀분석 연습문제 4.1 R 풀이
> 

- toc:true
- branch: master
- badges: true
- comments: true
- author: 김동준
- categories : [R, 회귀분석]

Base 코드

In [1]:
x <- c(28.0,28.0,32.5,39.0,45.9,57.8,58.1,62.5) #시간당평균온도
y <- c(12.4,11.7,12.4,10.8,9.4,9.5,8.0,7.5) #석탄소비량
lm31 <- lm(y~x)

# 1) 기울기와 절편의 신뢰구간 구하기

In [2]:
summary(lm31)


Call:
lm(formula = y ~ x)

Residuals:
    Min      1Q  Median      3Q     Max 
-0.5663 -0.4432 -0.1958  0.2879  1.0560 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 15.83786    0.80177  19.754 1.09e-06 ***
x           -0.12792    0.01746  -7.328  0.00033 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.6542 on 6 degrees of freedom
Multiple R-squared:  0.8995,	Adjusted R-squared:  0.8827 
F-statistic: 53.69 on 1 and 6 DF,  p-value: 0.0003301


In [4]:
coeffs <- coefficients(lm31)
confint(lm31,coeffs[2],level=0.95)  #기울기 b1 의 95% 신뢰구간

2.5 %      97.5 %     
x -0.1706383 -0.08520517

In [5]:
confint(lm31,level = 0.95) #절편 b0 의 95% 신뢰구간

2.5 %      97.5 %     
(Intercept) 13.8759886 17.79972621
x           -0.1706383 -0.08520517

# 2) 신뢰구간의 의미

	시간당 평균온도를 한 단위(℉) 늘리면 석탄소비량(ton)은 -0.171과 -0.0852 사이의 어느 한 값의 배수만큼 평균적으로 증가함을 신뢰계수 0.95로 예측할 수 있다.

# 3) 가설 검정 영가설=0 유의수준 0.05

In [6]:
summary(lm31)


Call:
lm(formula = y ~ x)

Residuals:
    Min      1Q  Median      3Q     Max 
-0.5663 -0.4432 -0.1958  0.2879  1.0560 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 15.83786    0.80177  19.754 1.09e-06 ***
x           -0.12792    0.01746  -7.328  0.00033 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.6542 on 6 degrees of freedom
Multiple R-squared:  0.8995,	Adjusted R-squared:  0.8827 
F-statistic: 53.69 on 1 and 6 DF,  p-value: 0.0003301


In [7]:
15.838 + 2.447* 0.01746
15.838 - 2.447* 0.01746

[1] 15.88072

[1] 15.79528

	기울기의 추정값 b1 = 15.838 이고 표준오차 SE는 0.01746 이다. 자유도(degrees of freedom)=6 인 t분포에서 가운데 면적이 95%인 t*=2.447 이며 95% 신뢰구간은 다음과 같이 계산한다.
	b_1±t^*×SE=15.838 ±2.447 ×0.01746
95% 신뢰구간 : (15.79528 , 15.88072) 

	이 결과로부터 기울기(평균온도가 1℉ 오를 때 석탄소비량 상승은 15.79528 ton 에서 15.88072 ton 사이에 있을 것이라고 95% 확신한다.


# 4) 가설검정2 

	Pr(>|t|) 값(p-value)의 값이 극히 작으므로 상당히 믿을 만한 정보이고,
	영가설 H0를 기각한다.
	즉, 기울기 β_1 은 의미있는 가정값이다.

# 5) 분산분석을 통한 가설검정

In [9]:
anova(lm31)

Df Sum Sq    Mean Sq   F value  Pr(>F)      
x         1  22.980816 22.980816 53.69488 0.0003300523
Residuals 6   2.567934  0.427989       NA           NA

	F-value는 53.695 이다. 분자의 자유도 1 과 6인 F-분포에서 p-value 는 0.00033301이다. 이 p값은 굉장히 작기 때문에 선형모형(기울기와 편차)의 가정이 좋은 가정이라고 말할 수 있다. 


# 6)시간당 평균 온도에 따른 소비량 추정, 신뢰구간 구하기

In [10]:
q6 <- data.frame(x=40)
predict(lm31,q6, level=0.95,interval="confidence") #신뢰구간

fit      lwr     upr     
1 10.72099 10.1301 11.31188

>	추정된 석탄소비량은 10.72099 ton 이다.
>	그 신뢰구간은 (10.1301, 11.31188) 이고 y값이 이 범위에 있을 확률이 95%라는 의미이다.


# 7) 예측구간 추정

In [11]:
predict(lm31,q6, level=0.95,interval="predict")예측구간

fit      lwr      upr     
1 10.72099 9.014624 12.42735

>	추정된 석탄소비량은 10.72099 ton 이다.
그 예측구간은 (9.014624, 12.42735) 이고 y값이 이 범위에 있을 확률이 95%라는 의미이다.

# 8) 6번 7번문제 비교

>	신뢰구간: y의 평균이 존재하는 구간을 추정

    	모집단을 예측 변수의 특정 값의 반응변수의 값으로 제한한다.
    
>	예측구간: 하나의 y가 존재하는 구간을 추정

    	모집단을 대부분 포함해야 하므로 구간이 더 넓다.

# 9) 결정계수 및 의미

In [12]:
cor(x,y) # 상관계수

[1] -0.9484139

>	상관 계수 r : -0.9484139

    	-1에 가까우므로 강한 음의 상관관계를 가진다고 볼 수 있다.


In [13]:
(cor(x,y))^2 #결정계수

[1] 0.8994889

	결정 계수 R^2: 0.8994889

결정 계수는 온도변화로 석탄소비량의 89.95%를 설명한다는 뜻이다.

결정 계수가 1에 거의 근접하므로 가정된 방정식이 온도변화와 석탄소비량의 관계를 잘 설명한다고 볼 수 있다.
